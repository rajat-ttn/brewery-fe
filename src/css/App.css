*{
  font-family: Proxima Nova, Arial, Helvetica, sans-serif;
}
body{ background-color: #f1f1f2;}
.wrapper{
  min-height:100%;
  position:relative;
  margin-bottom: 35px;
}
.msgStyle{
  text-align: center; width:100%; margin-top: 10px; font-weight: bold; margin-top: 20%;
}
.pageLayout{
  margin:0 30px;
}
.beerContainer{
  background-color: #e1e1e1;
  border-radius:8px;
  text-align: right;
  padding-top:25px;
  padding-right: 25px;
  width:100%;
  height:255px;
  color:#fff;
  background-image: url("../assets/images/container.png");
  background-repeat: no-repeat;
  background-size: 100% 120%;
  -webkit-box-shadow: -1px 1px 8px 2px #cccccc;
  -moz-box-shadow: -1px 1px 8px 2px #cccccc;
  box-shadow: -1px 1px 8px 2px #cccccc;
  position: relative;

}
.beertype{
  font: bold 28px/46px Proxima Nova, Arial, Helvetica, sans-serif;
}
.currentTemp{
  font: bold 65px/57px Proxima Nova, Arial, Helvetica, sans-serif;
  display:block;
}
.rangeTemp{
  font: bold 35px Proxima Nova, Arial, Helvetica, sans-serif;
  display:block;
}
.iconStyling{
  margin-top: -8px;
  margin-right:5px;
}
.tmpLabel{
  font: normal 14px Proxima Nova, Arial, Helvetica, sans-serif;
  display: block;
  margin-bottom: 15px;
}
.logoStyle{
  width: 41px;
  height: auto;
  float: left;
  margin-top: 6px;
  margin-left: 20px;
  margin-right: 6px;
}
span.currentTemp> span {
  font-size: 60px;
}
.beerContainer:hover{
  background-color: #f1f1f1;
  -webkit-box-shadow: -1px 1px 15px 5px #f1c40f;
  -moz-box-shadow: -1px 1px 15px 5px #f1c40f;
  box-shadow: -1px 2px 20px 5px #f1c40f;
}
header{
  background-color: #34302c;
  color:#ffffff;
  border-bottom: 2px solid #000000;
  box-shadow: 0 -2px 4px 2px #000000;
  -moz-box-shadow: 0 -2px 4px 2px #000000;
  -webkit-box-shadow: 0 -2px 4px 2px #000000;
}
header h1{
  font-size: 35px;
  font-weight: bold;
  padding-top: 8px;
}
.commonBoxModel{
  padding: 15px 15px;
}
.overlay {
  position: absolute;
  display: block;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: 8px;
  background-color: rgba(0,0,0,0.5);
  z-index: 2;
  padding-right: 25px;
  padding-top: 22px;
}
.redOverlay{
  background-color: rgba(1620,0,0,0.5);
}
.blueOverlay{
  background-color: rgba(0,0,1620,0.5);
}
.danger {
  -webkit-animation: 2s linear infinite condemned_blink_effect; // for android
  animation: 2s linear infinite condemned_blink_effect;
}
.red{color:#ff0000}
.blue{color:#025ab9}
.beerlistSection{
  padding-top: 20px;
}
.colorInfoWrapper{
  position:relative;
}
.colorInfo{
  top: 5px;
  right:8px;
  position: absolute;
}

.status-circle{
  width: 10px;
  height: 10px;
  border-radius: 5px;
  display: inline-block;
  margin-right: 5px;
  margin-left: 25px;
}
.blueBackground{background-color:#025ab9}
.redBackground{background-color:#ff0000}
@-webkit-keyframes condemned_blink_effect {
  0% {
    visibility: hidden;
  }
  20% {
    visibility: hidden;
  }
  100% {
    visibility: visible;
  }
}
@keyframes condemned_blink_effect {
  0% {
    visibility: hidden;
  }
  20% {
    visibility: hidden;
  }
  100% {
    visibility: visible;
  }
}
footer{
  background-color: #34302c;
  color:#ffffff;
  padding:5px;
  text-align: center;
  margin-top:15px;
  border-top:2px solid #aaaaaa;
  position:fixed;
  bottom:0;
  width:100%;
}
.dropdownSection{
  position: absolute;
  right: 100px;
  top: 18px;
}
.dropdownSection > select.form-control:not([size]):not([multiple]){
  height: 22px;
  padding-top: 0;
  padding-bottom: 0;
  font-size: 14px;
}
.bellIcon{
  width: 30px;
  height: 30px;
  position: absolute;
  right: 40px;
  top: 15px;
}
@media only screen and (max-width: 1024px){
  .beerContainer{
    height:290px;
  }
}
@media only screen and (max-width: 768px) {
  .beerContainer{
    height:270px;
  }
}
@media only screen and (max-width: 767px) {
  .beerContainer{
    height:245px;
  }
}
@media only screen and (max-width: 530px) {
  header {
    height:100px;
  }
  .bellIcon{
    top: 60px;
  }
  .dropdownSection{
    top: 60px;
  }
}
@media only screen and (min-width: 450px) and (max-width: 575px){
  .beerContainer{
    width:80%;
    margin:0 auto;
  }
}