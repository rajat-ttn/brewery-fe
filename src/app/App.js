import React, { Component } from 'react';
import { Provider } from 'react-redux';

import '../css/App.css';
import '../css/bootstrap.css';
import Dashboard from './components/Dashboard';
import store from './store';

class App extends Component {
    
    render() {
        return (
            <Provider store={store}>
                    <Dashboard />
            </Provider>
        );
    }
}

export default App;
