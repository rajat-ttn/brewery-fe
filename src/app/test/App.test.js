import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';


describe('App renders without crashing', () => {
    const initialState = {output:100};
    const mockStore = configureStore();
    let store = mockStore(initialState);


    // it('renders component correctly', () => {
    //     const div = document.createElement('div');
    //     ReactDOM.render(<App />, div);
    //     ReactDOM.unmountComponentAtNode(div);
    // });

    it('should render children correctly', () => {
        const appComponent = renderer.create(
            <Provider store={store}>
                <App />
            </Provider>
        );
        expect(appComponent.toJSON()).toMatchSnapshot();
    });
});

global.requestAnimationFrme = function(callback) {
    setTimeout(callback, 0);
};
