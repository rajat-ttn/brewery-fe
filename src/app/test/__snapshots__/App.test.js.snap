// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing should render children correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="loader"
  />
  <header>
    <img
      alt="logo"
      className="logoStyle"
      src="/images/beer_logo.png"
    />
    <h1>
      The Brewery
    </h1>
    <span
      onClick={[Function]}
    >
      <img
        alt="alarm"
        className="bellIcon"
        src="/images/alarm.png"
      />
    </span>
  </header>
  <div
    className="pageLayout"
  >
    <div
      className="dropdownSection"
    >
      <select
        className="input-sm form-control input-s-sm inline"
        onChange={[Function]}
        value="CELSIUS"
      >
        <option
          value="CELSIUS"
        >
          Celsius
        </option>
        <option
          value="FAHRENHEIT"
        >
          Fahrenheit
        </option>
      </select>
    </div>
    <div
      className="colorInfoWrapper"
    >
      <div
        className="colorInfo"
      >
        <span
          className="status-circle blueBackground"
        />
        <small>
          <strong>
            Too Low
          </strong>
        </small>
        <span
          className="status-circle redBackground"
        />
        <small>
          <strong>
            Too High
          </strong>
        </small>
      </div>
    </div>
    <div
      className="row beerlistSection"
    >
      <div
        className="msgStyle"
      >
        <p>
          No Container Found
        </p>
      </div>
    </div>
  </div>
  <footer>
    <small>
      Â© 2018 The Brewery  - Transport Refrigeration Sensors
    </small>
  </footer>
</div>
`;
